### PR TITLE
maint: add interfaces for L2 bridge/messaging contracts

### PR DESCRIPTION
**Pull request by smartcontracts**
First PR for adding interfaces for L2 contracts. Adds interfaces for all of the bridging and message passing contracts.

### CODE REVIEW

**Comment by tynes** [comment]
This needs another rebase @smartcontracts

**Comment by smartcontracts** [comment]
> This needs another rebase @smartcontracts

Rebase completed

File: packages/contracts-bedrock/semver-lock.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
   },
   "src/L1/L1ERC721Bridge.sol": {
     "initCodeHash": "0xda80d225e332f9ab38d5f3ae655df3eea1dd56dbb523036a25c78859d2931ada",
-    "sourceCodeHash": "0x0454bb88f7b52c362309935fa21ecc8be7bd6f0d888270b8a00677c60c1014cb"
```
Comment (tynes) [inline packages/contracts-bedrock/semver-lock.json:24]: nit: no semver bump in this contract

File: packages/contracts-bedrock/semver-lock.json [hunk 2]
```diff
@@ -93,7 +93,7 @@
   },
   "src/L2/L2CrossDomainMessenger.sol": {
     "initCodeHash": "0xe7090ce8119fe373d930baa68c1df585953b7f6e8beea2a0084589d394bcab64",
-    "sourceCodeHash": "0xea51d4b7a96c3a4c06d73c214208775b6845d5f696198c169ef7c14ea005d451"
```
Comment (tynes) [inline packages/contracts-bedrock/semver-lock.json:96]: nit: no semver bump in this contract

**Review by AmadiMichael — APPROVED** [review]
This LGTM

**Pull request merged**